src/app/tools/controllers.h

```diff
@@ -17,6 +17,94 @@ namespace app { namespace tools {
 
 using namespace gfx;
 
+// Adjustment for snap to isometric grid
+static void snap_isometric_line(ToolLoop* loop, Stroke& stroke, bool lineCtl)
+{
+  // Get last two points
+  Stroke::Pt& a = stroke[stroke.size() - 2];
+  Stroke::Pt& b = stroke[stroke.size() - 1];
+
+  // Get function invoked by line tool
+  bool lineTool = (string_id_to_brush_type(loop->getTool()->getId()) == kLineBrushType);
```

Comment:
Do we need to check the line tool or the line brush type? because here we are passing the tool ID to a function that receives a brush type (and I think it works only by chance because `"line"` is a brush type and a tool ID).

Probably you are looking for something like:
```suggestion
  const bool lineTool = (loop->getTool()->getId() == WellKnownTools::Line);
```
But you will need to add:
```c++
const char* WellKnownTools::Line = "line";
```
in `tool_box.cpp` and `.h`.